roles/jd-client/src/lib/template_receiver/mod.rs

```diff
@@ -62,8 +62,11 @@ impl TemplateRx {
             .expect("Invalid coinbase output in config");
         let stream = tokio::net::TcpStream::connect(address).await.unwrap();
 
-        let pub_key: Secp256k1PublicKey = authority_public_key;
-        let initiator = Initiator::from_raw_k(pub_key.into_bytes()).unwrap();
+        let initiator = match authority_public_key {
```

Comment:
Here you can turn `authority_public_key: Option<Secp256k1PublicKey>` in a variable `authority_public_key_x_only: Option<XOnlyPublicKey>` and initialize the initiator with `Initiator::new(autority_public_key_x_only)`. I understand that you didn't write the whole code, but perhaps the methods `Initiator::from_raw_k()` and `Initiator::without_pk()` can be eliminated.

Reply:
while I agree this would be a desirable improvement to the code, this would imply refactoring multiple parts of the codebase which don't really have much to do with the context of this PR

so in order to keep PR atomicity, I created this issue so we can implement this improvement in the future:
https://github.com/stratum-mining/stratum/issues/760